src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/__init__.py

```diff
@@ -32,4 +32,125 @@ def load_arguments(self, command):
+        from azure.cli.command_modules.storage.util import file_path_completer
+
+        path_help = 'The path to the file within the file share.'
+        if default_file_param:
+            path_help = '{} If the file name is omitted, the source file name will be used.'.format(path_help)
+        self.extra('path', options_list=options_list or ('--path', '-p'),
+                   required=default_file_param is None, help=path_help,
+                   validator=get_file_path_validator(default_file_param=default_file_param),
+                   completer=file_path_completer)
+        self.ignore('file_name')
+        self.ignore('directory_name')
+
+    def register_source_uri_arguments(self, validator, blob_only=False):
+        self.argument('copy_source', options_list=('--source-uri', '-u'), validator=validator, required=False,
+                      arg_group='Copy Source')
+        self.extra('source_sas', default=None, arg_group='Copy Source',
+                   help='The shared access signature for the source storage account.')
+        self.extra('source_container', default=None, arg_group='Copy Source',
+                   help='The container name for the source storage account.')
+        self.extra('source_blob', default=None, arg_group='Copy Source',
+                   help='The blob name for the source storage account.')
+        self.extra('source_snapshot', default=None, arg_group='Copy Source',
+                   help='The blob snapshot for the source storage account.')
+        self.extra('source_account_name', default=None, arg_group='Copy Source',
+                   help='The storage account name of the source blob.')
+        self.extra('source_account_key', default=None, arg_group='Copy Source',
+                   help='The storage account key of the source blob.')
+        if not blob_only:
+            self.extra('source_path', default=None, arg_group='Copy Source',
+                       help='The file path for the source storage account.')
+            self.extra('source_share', default=None, arg_group='Copy Source',
+                       help='The share name for the source storage account.')
+
+
+class StorageCommandGroup(AzCommandGroup):
+    def storage_command(self, name, method_name=None, command_type=None, **kwargs):
+        """ Registers an Azure CLI Storage Data Plane command. These commands always include the four parameters which
+        can be used to obtain a storage client: account-name, account-key, connection-string, and sas-token. """
+        command_name = self.command(name, method_name, command_type, **kwargs)
+        self.register_data_plane_account_arguments(command_name)
+
+    def storage_custom_command(self, name, method_name, **kwargs):
+        command_name = self.custom_command(name, method_name, **kwargs)
+        self.register_data_plane_account_arguments(command_name)
+
+    def register_data_plane_account_arguments(self, command_name):
```

Comment:
Please annotate this as an internal method with _.  I was under the impression you would have something like:

`g.register_data_plane_account_arguments('storage account')` and it made me very confused.